src/server/pfs/server/driver_v2.go

```diff
@@ -168,6 +172,32 @@ func (d *driverV2) withFileSet(ctx context.Context, repo, commit string, f func(
 	})
 }
 
+func (d *driverV2) withWriter(ctx context.Context, repo, commit string, f func(int64, *fileset.Writer) error) (retErr error) {
+	// TODO subFileSet will need to be incremented through postgres or etcd.
+	d.mu.Lock()
+	n := d.subFileSet
+	d.subFileSet++
+	d.mu.Unlock()
+
+	subFileSetStr := fileset.SubFileSetStr(n)
+	subFileSetPath := path.Join(repo, commit, subFileSetStr)
+	fsw := d.storage.NewWriter(ctx, path.Join(tmpPrefix, subFileSetPath))
+	defer func() {
+		if err := d.storage.Delete(ctx, path.Join(tmpPrefix, subFileSetPath)); retErr == nil {
+			retErr = err
+		}
+	}()
+	if err := f(n, fsw); err != nil {
```

Comment:
`f` -> `cb`. Also, if you see any callback code that uses the name `f`, go ahead and move it over to `cb`.